### DISCUSSION TITLE
很好奇这个网站的实现机制

### DISCUSSION THREAD

**Discussion by HifumiMizuhara** [share121/inter-knot#2118] [其他]
同上，想学习学习这样的网站是怎么做的，GitHub有开接口吗，还有为什么需要安装插件才能用

**Comment by chen-linkai** [comment]
首先是 Github API 是有提供且有相应文档的，例如 Github Docs 中提供了 [Github REST API](https://docs.github.com/zh/rest)。此处的油猴脚本相当于给网站“开后门”，默认的浏览器*为了用户安全*不允许**跨域**访问。这时候，想要访问 Github API 就被卡脖子了，因而此处作者 [share121](https://github.com/share121/) 选择使用一个脚本特别允许了在呈现出的网站内进行跨域访问。之后就是调用 API 以你的账号的身份访问这个仓库 [inter-knot](https://github.com/share121/inter-knot/)，这也解释了为什么进入网站要登录 Github 账号，一个原因就是 Github 的相关规定不允许个人账号多次滥用 API 接口，所以以我们自己的账号来访问被 Github 封禁的几率就小许多。利用 API ，你就可以通过 Javascript 获取仓库中 Diccussions 列表然后逐个访问内容最终呈现在网页即可。

**Reply by HifumiMizuhara** [reply]
谢谢高人指导，懂了

**Reply by share121** [reply]
我用的不是 Github REST API，我用的是 [Github GraphQL](https://docs.github.com/en/graphql/guides/using-the-graphql-api-for-discussions)

**Reply by HifumiMizuhara** [reply]
草，大佬出现
膜拜，学习了

**Reply by chen-linkai** [reply]
哈哈，我也没接触过 Github API ，用到的地方也只是 IDEA 令牌登录的时候见到过。此类网站在 Github 社区确实并不算少，作者可以的话也可以试试用 iframe 在帖子嵌套一个回复的区域试试看。跨域如果 DNS 服务器允许的话，是可以将 Github API 的域映射到自己的域下的~
